### PR TITLE
Parse http post 409, 401 status.

### PR DESCRIPTION
**Pull request by AndrewRAM33**
- add case http response status `409 Unauthorized` as logout and redirect to default
- add case http response status `409 Conflict` as system popup error
- refactor ErrorHook

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -53,38 +55,59 @@ function redirectOccurred(value: AxiosResponse<TeslerResponse>) {
...
+                }
+                break
+            }
+            case 409 : {
+                const notificationMessage = error.response.data.error?.popup?.[0] || ''
+                openButtonWarningNotification(
+                    notificationMessage,
+                    'OK',
+                    0,
+                    () => {Observable.empty()},
```
Comment (Dergash) [inline src/utils/api.ts:91]: Observables dObservable.empty is returned in our observables chains to terminate epic reaction to observed action.  
In this case nothing is returned and nothing is observed so it should be just `() => {}`

File: src/utils/api.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ import {ObjectMap, TeslerResponse} from '../interfaces/objectMap'
 import {$do} from '../actions/actions'
 import {OperationError} from '../interfaces/operation'
 import {BusinessError, ApplicationErrorType} from '../interfaces/view'
+import {openButtonWarningNotification} from './notifications'
+import {historyObj} from '..'
```
Comment (Dergash) [inline src/utils/api.ts:10]: I think it would be better if we explicitly declare on which module we are relying here

Comment (AndrewRAM33) [reply src/utils/api.ts]: Ok, changed path to '../reducers/router'

File: src/utils/api.ts (hunk 1 above)
Comment (AndrewRAM33) [reply src/utils/api.ts]: Fixed

**Pull request merged**